src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Exceptions/OpenIdConnectProtocolException.cs

```diff
@@ -26,17 +26,15 @@
 //------------------------------------------------------------------------------
 
 using System;
+using System.Runtime.Serialization;
 
 namespace Microsoft.IdentityModel.Protocols.OpenIdConnect
 {
 
-    #if DESKTOPNET45
-        [Serializable]
-    #endif
-
     /// <summary>
     /// This exception is thrown when an OpenIdConnect protocol handler encounters a protocol error.
     /// </summary>
+    [Serializable]
```

Comment:
w.r.t. Serializable, we need to make sure that the exceptions do serialize properly (tests).
Also, we need to account for additional properties (e.g. SecurityTokenExpiredException has Expires) by overriding GetObjectData #1300 

We should probably add [Serializable] attribute to SignedHttpRequest exceptions (we didn't set it as Serializable didn't work for other assemblies).

Reply:
I created a separate issue to add `[Serializable]` to `SignedHttpRequest` exceptions: #1324

I've also assigned both #1300 and #1324 to myself; I'll issue separate PRs for both.